Add CommitmentPrefix (#189)
cosmos/ics: spec/ics-003-connection-semantics/README.md

}
```

> Two helper functions are defined to provide automatic application of `CommitmentPrefix`. These functions are recommended to be used

Reviewer: Let's make them mandatory if we do it this way, otherwise the spec is ambiguous.

`CommitmentPrefix` can just be empty I presume.

Author: Yes `applyPrefix` can be a constant function.

> these functions should be used for introspecting other chains' state, instead of directly calling the `verifyMembership` with the clients.

Reviewer: ```suggestion
these functions MUST be used for introspecting other chains' state, instead of directly calling the `verifyMembership` or `verifyNonMembership` function on the client.
```
